### DISCUSSION TITLE
How to generate reports with custom template using mochawesome

### DISCUSSION THREAD

**Discussion by NeeruGana** [cypress-io/cypress#28533] [Questions and Help]
hi,
I am using Cypress to testing a web application and mochawesome for report generation, but i can see only one one format of html file I can generate. Is it possible to generate a report in another format(like for example in table form). And also I want to pass some data like name, id and pass/fail details from testScript to add to that report. I have researched and found that it may be possible with handlebars and giving reportTemplate as template file but its not working as expected. Any help on this is appreciated.
Thanks

**Comment by portyu9** [comment]
Hi @NeeruGana 

If you need a completely different report layout, such as your own table, I would not try to replace Mochawesome's HTML through `reportTemplate`.

The current supported approach is to have Mochawesome produce **JSON**, then render that JSON using your own template.

For example:

```js
const { defineConfig } = require('cypress')

module.exports = defineConfig({
  reporter: 'mochawesome',

  reporterOptions: {
    reportDir: 'cypress/results',
    overwrite: false,
    html: false,
    json: true,
  },
})
```

After the run you can merge the generated JSON files:

```bash
npx mochawesome-merge "cypress/results/*.json" -o mochawesome.json
```

You can then read `mochawesome.json` from your own Node script and generate whatever HTML structure you want with Handlebars, EJS, React, plain HTML, etc.

The JSON already contains test/suite status information such as passes and failures.

If you use `cypress-mochawesome-reporter` specifically and only want to attach extra metadata rather than redesign the whole report, it also supports adding test context, for example:

```js
cy.addTestContext(`ID: ${id}, Name: ${name}`)
```

So the distinction is:

- **Need extra information in the existing Mochawesome report:** add test context.
- **Need a completely different HTML/table design:** generate Mochawesome JSON and create your own renderer.

That is much more maintainable than trying to override the internals of `mochawesome-report-generator`.

Please mark this answer as accepted if it helped, thank you.
